src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Unix.cs

```diff
@@ -192,24 +193,22 @@ private void SendFileInternal(string? fileName, ReadOnlySpan<byte> preBuffer, Re
 
             // Open the file, if any
             // Open it before we send the preBuffer so that any exception happens first
-            FileStream? fileStream = OpenFile(fileName);
+            SafeFileHandle? fileHandle = OpenFileHandle(fileName);
 
             SocketError errorCode = SocketError.Success;
-            using (fileStream)
+
```

Comment:
Where is the SafeFileHandle getting disposed now?

Reply:
Well I missed it. `using` on a maybe-null handle is safe?

Reply:
Yes,  using null checks before calling Dispose